cellrank/tl/_utils.py

```diff
@@ -321,39 +332,224 @@ def _bias_knn(
     X
-        Matrix of `NxM` elements.
-    y:
-        Vector of `M` elements.
+        Array or sparse matrix of shape ``(n_cells, n_genes)`` containing the expression.
+    Y
+        Array of shape ``(n_cells, n_lineages)`` containing the absorption probabilities.
+    gene_names
+        Sequence of shape ``(n_genes,)`` containing the gene names.
+    method
+        Method for p-value calculation.
+    n_perms
+        Number of permutations if ``method='perm_test'``.
+    seed
+        Random seed if ``method='perm_test'``.
+    **kwargs
+        Keyword arguments for :func:`cellrank.ul._parallelize.parallelize`.
 
     Returns
     -------
-    :class:`numpy.ndarray`
-        The computed correlation.
+    :class:`pandas.DataFrame`
+        Dataframe of shape ``(n_genes, n_lineages * 5)`` containing the following columns, 1 for each lineage:
+
+            - ``{lineage} corr`` - correlation between the gene expression and absorption probabilities.
+            - ``{lineage} pval`` - calulated p-values for double-sided test.
+            - ``{lineage} qval`` - corrected p-values using Benjamini-Hochberg method at level `0.05`.
+            - ``{lineage} ci low`` - lower bound of the `95%` correlation confidence interval.
+            - ``{lineage} ci high`` - upper bound of the `95%` correlation confidence interval.
     """
 
-    X_bar, y_std, n = np.array(X.mean(axis=0)).reshape(-1), np.std(y), X.shape[0]
-    num = X.T.dot(y) - n * X_bar * np.mean(y)
-    denom = (
-        (n - 1) * np.std(X.A, axis=0) * y_std
-        if issparse(X)
-        else (X.shape[0] - 1) * np.std(X, axis=0) * y_std
+    corr, pvals, ci_low, ci_high = _correlation_test_helper(
+        X.T, Y.X, method=method, n_perms=n_perms, seed=seed, **kwargs
+    )
+    res = pd.DataFrame(corr, index=gene_names, columns=[f"{c} corr" for c in Y.names])
+    res[[f"{c} pval" for c in Y.names]] = pvals
+    res[[f"{c} qval" for c in Y.names]] = res[[f"{c} pval" for c in Y.names]].apply(
+        lambda c: multipletests(c, alpha=0.05, method="fdr_bh")[1]
     )
+    res[[f"{c} ci low" for c in Y.names]] = ci_low
+    res[[f"{c} ci high" for c in Y.names]] = ci_high
+
+    res = res[
+        [
+            f"{c} {stat}"
+            for c in Y.names
+            for stat in ("corr", "pval", "qval", "ci low", "ci high")
+        ]
+    ]
+    res.sort_values(by=[f"{c} corr" for c in Y.names], ascending=False, inplace=True)
 
-    if np.sum(denom == 0):
-        logg.warning(
-            f"No variation found in `{np.sum(denom==0)}` genes. Correlation for these will be `NaN`"
+    return res
+
+
+def _correlation_test_helper(
+    X: Union[np.ndarray, spmatrix],
+    Y: np.ndarray,
+    method: TestMethod = TestMethod.FISCHER,
+    n_perms: Optional[int] = None,
+    seed: Optional[int] = None,
+    **kwargs,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Compute the correlation between rows in matrix ``X`` columns of matrix ``Y``.
+
```

Comment:
Either here or somewhere else do we need to give a bit more meat: how do we compute these p-values and confidence intervals, where do the formulas come from? Need to cite something.

Reply:
Ican cite the original paper on Fischer transformation (from 1920s) to be shown in the docs.
If you want this internal only, I'd just add the link to wikipedia and that's it.